ddp-workspace/projects/ddp-dsm-ui/src/app/services/auth.service.ts

```diff
@@ -46,6 +46,14 @@ export class Auth {
 
   selectedStudy = new BehaviorSubject<string>('');
 
+  storageEventSource$ = fromEvent<StorageEvent>(window, 'storage').subscribe(
+    event => {
+      if (event.key === SessionService.DSM_TOKEN_NAME && event.newValue === null) {
```

Comment:
**[OPTIONAL]**
What do you think about moving the subscription part to the constructor function?
Something like this:
```
class authService {

  storageEventSourceSubscription$: Subscription;


 constructor() {
     this.storageEventSourceSubscription$ = fromEvent<StorageEvent>(window, 'storage').subscribe(
      event => {
        if (event.key === SessionService.DSM_TOKEN_NAME && event.newValue === null) {
          this.doLogout();
          this.router.navigateByUrl('/');
        }
      }
    );
  }
}
```

Reply:
Or to the separate function, which will be called in the constructor